packages/x-date-pickers/src/internals/index.ts

```diff
@@ -11,8 +11,12 @@ export type {
   MobileWrapperSlotsComponentsProps,
 } from './components/wrappers/MobileWrapper';
 export { MobileKeyboardInputView } from './components/CalendarOrClockPicker/CalendarOrClockPicker';
-export { PickersArrowSwitcher } from './components/PickersArrowSwitcher';
-export type { ExportedArrowSwitcherProps } from './components/PickersArrowSwitcher';
+export { PickersArrowSwitcher } from './components/PickersArrowSwitcher/PickersArrowSwitcher';
+export type {
+  ExportedPickersArrowSwitcherProps,
+  PickersArrowSwitcherSlotsComponent,
+  PickersArrowSwitcherSlotsComponentsProps,
+} from './components/PickersArrowSwitcher';
```

Comment:
Nitpick: Wouldn't just doing this achieve the same here as it already exports everything that is mentioned here in the `index.ts`.
Or do we want to be so explicit/in–control? 🤔 
```suggestion
export * from './components/PickersArrowSwitcher';;
```

Reply:
In `internal` I am always very explicit
But maybe that's not super relevant